### PR TITLE
[AC-2646] Remove FC MVP dead code from Core

### PR DESCRIPTION
**Pull request by vincentsalucci**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/AC-2646

## 📔 Objective

> Remove FC MVP dead code from the following: `CreateGroupCommand`, `UpdateGroupCommand`, `UpdateOrganizationUserCommand`, `OrganizationService`, `CollectionService`, `CipherService`

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by eliykat — CHANGES_REQUESTED** [review]
A few code review comments, I have also fixed various tests here: https://github.com/bitwarden/server/pull/4467

File: test/Core.Test/Vault/Services/CipherServiceTests.cs [hunk 1]
```diff
@@ -25,62 +25,6 @@ namespace Bit.Core.Test.Services;
...
-                        cols.All(c => collections.Any(x => c.Name == x.Name))),
-            Arg.Is<IEnumerable<CollectionCipher>>(c => c.Count() == ciphers.Count),
-            Arg.Is<IEnumerable<CollectionUser>>(i => !i.Any()));
-        await sutProvider.GetDependency<IPushNotificationService>().Received(1).PushSyncVaultAsync(importingUserId);
-        await sutProvider.GetDependency<IReferenceEventService>().Received(1).RaiseEventAsync(
-            Arg.Is<ReferenceEvent>(e => e.Type == ReferenceEventType.VaultImported));
-    }
-
     [Theory, BitAutoData]
     public async Task ImportCiphersAsync_IntoOrganization_WithFlexibleCollectionsEnabled_Success(
```
[the remark this answers is not in the document]
Comment (vincentsalucci) [reply test/Core.Test/Vault/Services/CipherServiceTests.cs:29]: `WithFlexibleCollectionsEnabled`
`WithFlexibleCollections`
`FlexibleCollectionsDisabled`
`WithoutFlexibleCollections`

**Review by r-tome — APPROVED** [review]

**Review by eliykat — DISMISSED** [review]
Nice work 🎉

**Review by shane-melton — APPROVED** [review]
Vault changes look good!

**Comment by vincentsalucci** [comment]
@eliykat @shane-melton @r-tome Sorry ya'll - had to fix some conflicts!

**Review by eliykat — APPROVED** [review]

**Review by shane-melton — APPROVED** [review]

**Pull request merged**